src/pages/fieldManagement/settings/defaultFieldComponents/relationshipComponents/Type.jsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import { get, omit } from 'lodash-es';
+import { v4 as uuid } from 'uuid';
+
+import InputAdornment from '@material-ui/core/InputAdornment';
+import IconButton from '@material-ui/core/IconButton';
+import NewChildIcon from '@material-ui/icons/AddCircle';
+
+import TextInput from '../../../../../components/inputs/TextInput';
+import DeleteButton from '../../../../../components/DeleteButton';
+import Role from './Role';
+
+function deleteType(relationships, typeGuid) {
+  return omit(relationships, [typeGuid]);
+}
+
+function updateType(relationships, typeGuid, newTypeLabel) {
+  const type = get(relationships, typeGuid, {});
+  return {
+    ...relationships,
+    [typeGuid]: {
+      ...type,
+      label: newTypeLabel,
+    },
+  };
+}
+
+function addRole(relationships, typeGuid) {
```

Comment:
Validation rather than post-hoc de-duping was definitely the cleaner choice here. Nice.